pkg/apiserver/handlers.go

```diff
@@ -281,17 +307,40 @@ func (tw *baseTimeoutWriter) timeout(msg string) {
 			enc := json.NewEncoder(tw.w)
 			enc.Encode(errors.NewServerTimeout(api.Resource(""), "", 0))
 		}
+	} else {
+		// The timeout writer has been used by the inner handler. There is
+		// no way to timeout the HTTP request at the point. We have to shutdown
+		// the connection for HTTP1 or reset stream for HTTP2.
+		//
+		// Note from: Brad Fitzpatrick
```

Comment:
The original crash was caused because panics weren't being handed back up to the HTTP server to recover. I'm fine with a panic here, because it would be handed back to the server. 

We should invert the nesting of timeout and `RecoverPanics` so that we are again protected from panic crashes.